### PR TITLE
fixes for ManagedWebSocket server mode

### LINKED ISSUE #10490: ManagedWebSocket doesn't send payloads if `_isServer` is true
Looks like [`System.Net.WebSockets.ManagedWebSocket.WriteFrameToSendBuffer`](https://github.com/dotnet/corefx/blob/b835f230039c4f44944b7fb3f8ce670ebf578ea4/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ManagedWebSocket.cs#L448) fails to actually write the payload if `_isServer` is set. It copies the payload into the send buffer in the same if-branch that handles masking (which is client-only :smile:).

Discovered and fixed this while porting code to ASP.NET, so I'm going to port the fix over to corefx and open a PR.

/cc @stephentoub @Tratcher

### PR DESCRIPTION
**Pull request by anurse**
Updates to ManagedWebSocket to fix issues in server mode. There isn't really a place to put tests in corefx since no public API exercises this code path (we copy this class as raw source code into ASP.NET right now, as a stop-gap to a proper public API existing). I do have got tests for these changes in my copy of the WebSocket code, currently in PR at https://github.com/aspnet/WebSockets/pull/105.

As far as breaking corefx, the only real concern is if the existing tests pass, since the new code path is not ever used by a public corefx API (I just want to keep our copy and the copy in corefx in sync).

/cc @Petermarcu (since @stephentoub is OOF, can you suggest someone to take a look at this?)
/cc @Tratcher 

Fixes #10490

### CODE REVIEW

File: src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ManagedWebSocket.cs [hunk 1]
```diff
@@ -463,15 +464,20 @@ private int WriteFrameToSendBuffer(MessageOpcode opcode, bool endOfMessage, Arra
             {
                 // We need to know where the mask starts so that we can use the mask to manipulate the payload data,
                 // and we need to know the total length for sending it on the wire.
-                int maskOffset = WriteHeader(opcode, _sendBuffer, payloadBuffer, endOfMessage, useMask: true);
-                headerLength = maskOffset + MaskLength;
+                maskOffset = WriteHeader(opcode, _sendBuffer, payloadBuffer, endOfMessage, useMask: true);
+                headerLength = maskOffset.Value + MaskLength;
```
Comment (stephentoub) [inline src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ManagedWebSocket.cs]: Slightly cheaper as GetValueOrDefault; we know it has a value s we just assigned one.

**Comment by stephentoub** [comment]
One nit, otherwise LGTM

**Comment by stephentoub** [comment]
> There isn't really a place to put tests in corefx since no public API exercises this code path

We can add some unit tests around this with a minimal test-only server that pulls this file into the tests. I can look at that when I'm actually back online (pretend I'm not here :-)

File: src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ManagedWebSocket.cs (hunk 1 above)
Comment (anurse) [inline src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ManagedWebSocket.cs]: It seems mildly odd that is the case, but looking at the code for `Nullable<T>` I can see why :). Will do!

**Comment by anurse** [comment]
That's basically what I was thinking but I wasn't familiar enough with the repo structure to get much progress on that. Thanks for hopping online quick for this @stephentoub !

**Comment by anurse** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please

**Comment by anurse** [comment]
Looks like this is passing tests. I'm not sure exactly what the merge procedure is though. Can someone help merge this in?

**Comment by mellinoe** [comment]
LGTM. We can add the tests that @stephentoub mentioned if/when we start using this in public corefx API.

**Pull request merged**